### ISSUE TITLE
[Bug] Continual Training by only loading peft adapter

### ISSUE THREAD

**Issue by 14H034160212** [OpenGVLab/InternVL#689]
### Checklist

- [X] 1. I have searched related issues but cannot get the expected help.
- [X] 2. The bug has not been fixed in the latest version.
- [X] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.

### Describe the bug

The bug is to load the trained peft adapter and do continual training and the loss cannot decrease exactly following the loss in the previous round. If we load the saved deepspeed checkpoint, the model will decrease the loss exactly following the previous round but if we only save the adapter model, the loss will not decrease exactly same as the previous round.

### Reproduction

```
HF_DATASETS_CACHE="/mnt/share" \
MMDOC_DATA_DIR="/mnt/share/data" \
HUGGING_FACE_HUB_TOKEN="XXX" \
MMDOC_CACHE_HOME="/mnt/share/qiming" \
CUDA_VISIBLE_DEVICES=0 \
torchrun \
  --nnodes=1 \
  --node_rank=0 \
  --master_addr=127.0.0.1 \
  --nproc_per_node=1 \
  --master_port=34229 \
  /home/paperspace/mmdoc/projects/InternVL2/internvl_chat/internvl/train/internvl_chat_finetune.py \
  --model_name_or_path OpenGVLab/InternVL2-1B \
  --conv_style Hermes-2 \
  --output_dir /home/paperspace/mmdoc/outputs/internvl/Desilian_WF69_20240816_ppromtV2 \
  --meta_path Desilian_WF69_2262_Learning_Docs_20240816 \
  --force_image_size 448 \
  --max_dynamic_patch 6 \
  --down_sample_ratio 0.5 \
  --drop_path_rate 0.0 \
  --freeze_llm True \
  --freeze_mlp True \
  --freeze_backbone True \
  --use_llm_lora 16 \
  --vision_select_layer -1 \
  --dataloader_num_workers 8 \
  --bf16 True \
  --num_train_epochs 10 \
  --per_device_train_batch_size 6 \
  --gradient_accumulation_steps 1 \
  --evaluation_strategy no \
  --save_strategy steps \
  --save_steps 200 \
  --save_total_limit 1 \
  --learning_rate 30e-5 \
  --weight_decay 0.01 \
  --warmup_ratio 0.03 \
  --lr_scheduler_type cosine \
  --logging_steps 10 \
  --max_seq_length 3072 \
  --do_train True \
  --grad_checkpoint True \
  --group_by_length False \
  --dynamic_image_size True \
  --use_thumbnail True \
  --ps_version v2 \
  --deepspeed /home/paperspace/mmdoc/projects/InternVL2/internvl_chat/zero_stage1_config.json \
  --report_to wandb \
  --is_Xtracta_dataset True \
  --header_only True \
  --project_name Desilian_WF69_2262_Learning_Docs_20240816-InternVL2.0 \
  --add_page_promt True \
  --save_only_model
```

### Environment

```Shell
Using the InternVL2 package environment
```
